pkg/server/storage_api/storage_api_test: TestAdminDecommissionedOperations failed

pkg/server/storage_api/storage_api_test.TestAdminDecommissionedOperations [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11124353?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11124353?buildTab=artifacts#/) on master @ [e57e9742146b7fac7f7530fa6224e11121ada92f](https://github.com/cockroachdb/cockroach/commits/e57e9742146b7fac7f7530fa6224e11121ada92f):


```
=== RUN   TestAdminDecommissionedOperations
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/800529510c8a77f305555fe4848c57f0/logTestAdminDecommissionedOperations246959236
    test_log_scope.go:81: use -show-logs to present logs inline
=== CONT  TestAdminDecommissionedOperations
    decommission_test.go:978: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/800529510c8a77f305555fe4848c57f0/logTestAdminDecommissionedOperations246959236
--- FAIL: TestAdminDecommissionedOperations (21.23s)
=== RUN   TestAdminDecommissionedOperations/Drain
    decommission_test.go:973: 
        	Error Trace:	pkg/server/storage_api/storage_api_test_test/pkg/server/storage_api/decommission_test.go:973
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				pkg/server/storage_api/storage_api_test_test/pkg/server/storage_api/decommission_test.go:955
        	Error:      	Not equal: 
        	            	expected: 0x2
        	            	actual  : 0x4
        	Test:       	TestAdminDecommissionedOperations/Drain
        	Messages:   	rpc error: code = DeadlineExceeded desc = context deadline exceeded
    --- FAIL: TestAdminDecommissionedOperations/Drain (10.00s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminDecommissionedOperations.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30237

--- a/pkg/server/storage_api/decommission_test.go
+++ b/pkg/server/storage_api/decommission_test.go
@@ -785,6 +785,9 @@ func TestAdminDecommissionedOperations(t *testing.T) {
 	})
 	defer tc.Stopper().Stop(ctx)
 
+	// Configure drain to immediately cancel SQL queries and jobs to speed up the
+	// test and avoid timeouts.
+	serverutils.SetClusterSetting(t, tc, "server.shutdown.query_wait", 0)
 	serverutils.SetClusterSetting(t, tc, "server.shutdown.jobs_wait", 0)
 
 	scratchKey := tc.ScratchRange(t)
@@ -801,8 +804,8 @@ func TestAdminDecommissionedOperations(t *testing.T) {
 		require.NoError(t, srv.Decommission(ctx, status, []roachpb.NodeID{decomSrv.NodeID()}))
 	}
 
-	testutils.SucceedsWithin(t, func() error {
-		timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	testutils.SucceedsSoon(t, func() error {
+		timeoutCtx, cancel := context.WithTimeout(ctx, testutils.DefaultSucceedsSoonDuration)
 		defer cancel()
 		_, err := decomSrv.DB().Scan(timeoutCtx, keys.LocalMax, keys.MaxKey, 0)
 		if err == nil {
@@ -813,7 +816,7 @@ func TestAdminDecommissionedOperations(t *testing.T) {
 			return nil
 		}
 		return err
-	}, 10*time.Second)
+	})
 
 	// Set up an admin client.
 	adminClient := decomSrv.GetAdminClient(t)
@@ -925,8 +928,8 @@ func TestAdminDecommissionedOperations(t *testing.T) {
 	for _, tc := range testcases {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
-			testutils.SucceedsWithin(t, func() error {
-				timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+			testutils.SucceedsSoon(t, func() error {
+				timeoutCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
 				defer cancel()
 				err := tc.op(timeoutCtx, adminClient)
 				if tc.expectCode == codes.OK {
@@ -945,7 +948,7 @@ func TestAdminDecommissionedOperations(t *testing.T) {
 				}
 				require.Equal(t, tc.expectCode, s.Code(), "%+v", err)
 				return nil
-			}, 10*time.Second)
+			})
 		})
 	}
 }

